src/components/hero/interactive.tsx

```diff
@@ -0,0 +1,207 @@
+import React from "react";
+
+import { NotificationManager } from "react-notifications";
+import { serveFile } from "./utils";
+import {
+  InputFileSpecification,
+  ColumnSpecification,
+  BackButton,
+  SubmitButton,
+} from "./subcomponents";
+
+interface InteractiveTerminalProps {
+  onClick: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
+}
+
+interface InteractiveTerminalState {
+  column_id: string;
+  column_value: string;
+  column_sort: string;
+  column_kind: string;
+  data_file: string;
+  loading: boolean;
+}
+
+export default class InteractiveTerminal extends React.Component<
+  InteractiveTerminalProps,
+  InteractiveTerminalState
+> {
```

Comment:
I guess this could also be written as a FunctionalComponent. But I guess it's also fine to keep is as a class since it's quite complex.